distribution/src/root/config/hazelcast-hardened.yaml

```diff
@@ -0,0 +1,233 @@
+# This is a sample Hazelcast configuration focused on hardened security.
+hazelcast:
+  # Custom cluster name protects against unintended cluster formation between foreign members.
+  cluster-name: ${custom.cluster.name}
+  properties:
+    # Don't listen on all network interfaces, but bind to just one.
+    hazelcast.socket.bind.any: false
+
+  management-center:
+    # Explicitly disable the scripting operations sent from the Management center.
+    # The default value is false already so this one is just to be sure and explicit.
+    scripting-enabled: false
+
+  user-code-deployment:
+    # Explicitly disable the user-code deployments. I.e. allow only code which already is on the classpath.
+    # The default value is false already so this one is just to be sure and explicit.
+    enabled: false
+
+  jet:
+    # If you don't need stream/batch processing features, leave them disabled.
+    enabled: false
+    # If you allow the Jet jobs, think twice before allowing custom code upload to the cluster!
+    resource-upload-enabled: false
+
+  # Use the Advanced networking feature to separate protocols used in Hazelcast. Each protocol will have its own endpoint.
+  # Protection introduced by the Advanced networking feature can be simply extended by the operating system features (e.g. firewall).
+  advanced-network:
+    enabled: true
+    join:
+      # Safer than the autodetection is the explicit usage of one discovery method. Let's disable the autodetection.'
+      auto-detection:
+        enabled: false
+      # If you don't need to, don't use the multicast discovery as it advertises own socket address and cluster configuration
+      # within the network
+      multicast:
+        enabled: false
+      # The TCP-IP discovery directly lists explicitly cluster member addresses.
+      tcp-ip:
+        enabled: true
+        # Use the member addresses including the port numbers, so Hazelcast won't try other socket addresses as a fallback.'
+        member-list:
+          - 192.168.1.101:15701
```

Comment:
What is the user supposed to do when they don't know the IP(s) in advance? Do you expect some external scripting tool/variable replacement to handle that?

And what are they supposed to do when the Hz members share the network with the application, e.g. 10.0.0.*, and can get any IP from that range?

Reply:
As this is just an example I would let users rewrite manually the configuration to fit their environment. This file should just emphasize places that can help with securing the installation.